packages/jest-mock/src/index.ts

```diff
@@ -1359,14 +1363,18 @@ export class ModuleMocker {
       descriptor = Object.getOwnPropertyDescriptor(proto, propertyKey);
       proto = Object.getPrototypeOf(proto);
     }
-    if (!descriptor) {
+    if (!descriptor && !options?.tolerateUndefined) {
       throw new Error(`${String(propertyKey)} property does not exist`);
     }
-    if (!descriptor.configurable) {
+
+    if (descriptor?.configurable === false) {
       throw new Error(`${String(propertyKey)} is not declared configurable`);
     }
+    if (descriptor?.writable === false) {
```

Comment:
`writable` check was missing. I bump into it while playing with this case:

https://github.com/facebook/jest/blob/bb39cb2c617a3334bf18daeca66bd87b7ccab28b/packages/babel-jest/src/__tests__/getCacheKey.test.ts#L195-L197

Hm.. What to do with non-writable values like `process.version`? Would be nice to replace them too. Perhaps via `tolerateFrozen` option?